filebeat/input/v2/input.go

```diff
@@ -118,6 +118,22 @@ func (c *Context) UpdateStatus(status status.Status, msg string) {
 	}
 }
 
+// MetricsRegistryOverrideID sets the "id" variable in the Context's
+// MetricsRegistry and returns the modified registry. This is required as some
+// inputs do not use their input ID as the identifier for their metrics.
+func (c *Context) MetricsRegistryOverrideID(id string) *monitoring.Registry {
```

Comment:
Please make these standalone helpers taking a registry as a parameter (and returning nothing) rather than methods on Context, there's no reason Context itself should have to be passed through to override a registry parameter.